md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDependencyDAO.java

```diff
@@ -83,16 +84,16 @@
 
 
                 //Listing downstream processes
-                Criteria checkDownstreamParentProcesses = session.createCriteria(Process.class).add(Restrictions.eq("enqueuingProcessId", parentProcessId)).add(Restrictions.eq("deleteFlag", false)).setProjection(Projections.property("process"));
+                Criteria checkDownstreamParentProcesses = session.createCriteria(Process.class).add(Restrictions.eq("enqueuingProcessId", parentProcessId)).add(Restrictions.eq(DELETE_FLAG, false)).setProjection(Projections.property("process"));
                 List<Integer> downstreamParentProcessIdList = new ArrayList<Integer>();
-                if (checkDownstreamParentProcesses.list().size() != 0) {
+                if (!checkDownstreamParentProcesses.list().isEmpty()) {
                     List<Process> downstreamParentProcessList = checkDownstreamParentProcesses.list();
                     for (Process p : downstreamParentProcessList) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)